morf-core/src/main/java/org/alfasoftware/morf/jdbc/ResultSetIterator.java

```diff
@@ -85,7 +104,9 @@ public ResultSetIterator(Table table, String query, Connection connection, SqlDi
     try {
       this.statement = connection.createStatement(ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
       this.statement.setFetchDirection(ResultSet.FETCH_FORWARD);
-      this.statement.setFetchSize(sqlDialect.fetchSizeForBulkSelects());
+      this.statement.setFetchSize((connectionResources.isPresent() && connectionResources.get().getFetchSizeForBulkSelects() != null) ?
+          connectionResources.get().getFetchSizeForBulkSelects() : sqlDialect.fetchSizeForBulkSelects());

```

Comment:
`connectionResources.map(ConnectionResources::getFetchSizeForBulkSelects)
    .orElse(sqlDialect.fetchSizeForBulkSelects())`

That will also handle the null-returning `getFetchSizeForBulkSelects()` for you automatically :)